hazelcast/src/main/java/com/hazelcast/client/impl/protocol/AuthenticationBaseMessageTask.java

```diff
@@ -0,0 +1,95 @@
+package com.hazelcast.client.impl.protocol;
+
+import com.hazelcast.client.AuthenticationException;
+import com.hazelcast.client.ClientEndpoint;
+import com.hazelcast.client.impl.client.ClientPrincipal;
+import com.hazelcast.client.impl.operations.ClientReAuthOperation;
+import com.hazelcast.instance.MemberImpl;
+import com.hazelcast.instance.Node;
+import com.hazelcast.nio.Address;
+import com.hazelcast.nio.Connection;
+import com.hazelcast.security.Credentials;
+import com.hazelcast.util.UuidUtil;
+
+import java.security.Permission;
+import java.util.Collection;
+import java.util.Set;
+
+/**
+ * Base authentication task
+ */
+public abstract class AuthenticationBaseMessageTask<CM extends ClientMessage>
```

Comment:
Wouldn't it be more like BaseAuthenticationMessageTask or AbstractAuthenticationMessageTask?